src/Microsoft.AspNetCore.Rewrite/Internal/CodeRules/RedirectRule.cs

```diff
@@ -38,9 +48,17 @@ public override void ApplyRule(RewriteContext context)
             {
                 var newPath = initMatchResults.Result(Replacement);
                 var response = context.HttpContext.Response;
+
                 response.StatusCode = StatusCode;
+                context.Result = RuleTermination.ResponseComplete;
+
+                if (string.IsNullOrEmpty(newPath))
```

Comment:
Since we are speculating on this behavior, could you add a test case to show the valid behavior that hits this path? Otherwise, let's thread this as an error condition instead of redirecting to '/'